v1-2/learn/how-to-test-ballerina-code.md

```diff
@@ -403,3 +404,74 @@ function testService() {
     }
 }
 ```
+
+## Function Mocks
+Testerina provides the functionality to mock a function in a different thrird-party module with your own Ballerina function, which will help you test your Ballerina module independently.
+
+#### @test:Mock {}
+
+The function specified with the `@test:Mock {}` annotation will be considered as a mock function that gets triggered every time the original function is called. The original function that will be mocked should be defined using the annotation parameters.
+
+##### Annotation Value Fields
+
+`moduleName : "<moduleName>"` : Name of the module where the function to be mocked resides in. If the function is within the same module, this can be left blank or "." (No module) can be passed. If the function is in a different module, but within the same project, just passing the module name will suffice. For functions in completely seperate modules, the fully qualified module name must be passed, which includes the `orgName` and the `version`. ie. `orgName/module:version`. For native function, the ballerina module needs to be specified. 
+
+`functionName : "<functionName>"` : Name of the function to be mocked.
+
+Sample :
+
+The following is an example for function mocking.
+
+```ballerina
+import ballerina/io;
+import ballerina/test;
+import ballerina/math;
+
+
+// This is the mock function which will replace the real `intAdd` function
+@test:Mock {
+    // Since the function is defined in the same module, "." can be passed as the current module.
+    // This can also be left blank.
+    moduleName : ".",
+    functionName : "intAdd"
+}
+// The mock function signature should match the actual function signature.
+public function mockIntAdd(int a, int b) returns (int) {
+    io:println("I am the mockIntAdd function");
+    return (a - b);
+}
+
+
+// This test function calls the local `intAdd()` function but it expects the mocked result
+@test:Config {}  
+function test_intAdd() {
+    int answer = 0;
+    answer = intAdd(5, 3);
+    test:assertEquals(answer, 2, msg = "function mocking failed");
+}
+
+public function intAdd(int a, int b) returns (int) {
```

Comment:
better to split the sample in to actual files. ie intAdd in the module and mock in the test file. since we only support mocking in projects

Reply:
I split the function into a separate definition and I have mentioned that the function in the file is the one we are trying to mock